src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs

```diff
@@ -467,11 +467,11 @@ protected FormattableString GetResourceTypeExpression(ResourceTypeSegment resour
             if (resourceType == ResourceTypeSegment.ResourceGroup)
                 return $"{typeof(ResourceGroup)}.ResourceType";
```

Comment:
Therefore we decide not to add `Resource` suffix to `ResourceGroup`?

Reply:
Yes and now.

- Previously it's decided that we will not add `Resource` suffix to the following 3 types:
  - `ResourceGroup`
  - `ResourceProvider` (aka `Provider`, but renamed to `ResourceProvider`)
  - `ResourceLink`
- After the latest discussion, we decided to unify the naming pattern, so all resource types should end with `Resource`.

I'll update this PR to change according to the latest disucssion.

Reply:
So the conclusion is that we will still have
- ResourceGroupResource
- ResourceProviderResource
- ResourceLinkResource

Reply:
@ArcturusZhang  That's correct. All now ends with `Resource`.